graylog2-web-interface/src/components/users/UserCreate/UserCreate.jsx

```diff
@@ -40,10 +42,19 @@ import UsernameFormGroup from './UsernameFormGroup';
 
 import { Headline } from '../../common/Section/SectionComponent';
 
+const isCloud = AppConfig.isCloud();
+const cloudPlugin = isCloud ? PluginStore.exports('cloud') : null;
+
 const _onSubmit = (formData, roles, setSubmitError) => {
-  const data = { ...formData, roles: roles.toJS(), permissions: [] };
+  let data = { ...formData, roles: roles.toJS(), permissions: [] };
   delete data.password_repeat;
-  data.username = data.username.trim();
+
+  if (isCloud && cloudPlugin) {
+    const onCreate = cloudPlugin?.[0]?.oktaUserForm.onCreate;
+    data = onCreate(data);
```

Comment:
I think using the optional chaining operator this way does not help us: Using the operator only in the first two accesses would mean that if one of them is `null` or `undefined`, the error thrown will just occur on the next level (when accessing `onCreate` in this case).

Since using a cloud instance will also use Okta for authentication and user management, I think it's safe to assume that a cloud instance will also have the cloud plugin installed, and not having it is an error itself. Rendering the regular user form in case the plugin is not there for some reason is also wrong, since it disguises an error that would lead to possible data inconsistencies and other failures when sending data to Okta.

From that perspective, my opinion is that we should assume that a cloud instance will always have the `cloud` plugin installed. Therefore, I think it is fine to change things a bit to make code more readable, starting with this change in line 46:

```diff
- const cloudPlugin = isCloud ? PluginStore.exports('cloud') : null; 
+ const oktaUserForm = isCloud ? PluginStore.exports('cloud')[0].oktaUserForm : null; 
```

That would let us access properties of that interface directly in other places, for instance on this block:
```diff
-  if (isCloud && cloudPlugin) {
-    const onCreate = cloudPlugin?.[0]?.oktaUserForm.onCreate;
-    data = onCreate(data);
+  if (isCloud) {
+    const { onCreate } = oktaUserForm;
+    data = onCreate(data);
```

We could also try to display a proper error message if the plugin is not available, for that I would enclose line 46 in a `try catch` statement and handle that situation to display an error message instead of the form in that case.